File: BOJ/Gold/11657 타임머신/Main.java
import java.io.*;
import java.util.*;

public class Main {
    static final long INF = Long.MAX_VALUE;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] firstLine = br.readLine().split(" ");
        int n = Integer.parseInt(firstLine[0]);
        int m = Integer.parseInt(firstLine[1]);

        int[][] edges = new int[m][3];
        for (int i = 0; i < m; i++) {
            String[] edgeLine = br.readLine().split(" ");
            edges[i][0] = Integer.parseInt(edgeLine[0]);
            edges[i][1] = Integer.parseInt(edgeLine[1]);
            edges[i][2] = Integer.parseInt(edgeLine[2]);
        }

        long[] result = ballmanFord(1, n, m, edges);

        if (result == null) {
            System.out.println(-1);
        } else {
            for (int i = 2; i <= n; i++) {
                if (result[i] != INF)
                    System.err.println(result[i]);
                else System.err.println(-1);
            }
        }

    }
    
    public static long[] ballmanFord(int start, int n, int m, int[][] edges) {
        long[] dist = new long[n + 1];
        Arrays.fill(dist, INF);
        dist[start] = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                int u = edges[j][0];
                int v = edges[j][1];
                int w = edges[j][2];

                if (dist[u] != INF && dist[v] > dist[u] + w) {
                    dist[v] = dist[u] + w;

                    if (i == n - 1) {
                        return null;
                    }

                }
            }
        }
        return dist;
    }
}

File: BOJ/Gold/11657 타임머신/main.py
# https://www.acmicpc.net/problem/11657
INF = float('inf')

n, m = map(int, input().split())
edges = []
for _ in range(m):
    edges.append(list(map(int, input().split())))

def bellman_ford(start, n, edges):
    dist = [INF] * (n+1)
    dist[start] = 0
    negative_cycle = False

    for i in range(n):
        for u, v, w in edges:
            if dist[u] != INF and dist[v] > dist[u] + w:
                dist[v] = dist[u] + w
                if i == n -1 :
                    negative_cycle = True

    if negative_cycle:
        print(-1)
    else:
        for i in range(2, n+1):
            if dist[i] == INF:
                print(-1)
            else:
                print(dist[i])

bellman_ford(1, n, edges)
